source/common/router/upstream_request.h

```diff
@@ -130,6 +180,9 @@ class UpstreamRequest : public Logger::Loggable<Logger::Id::router>,
   bool hadUpstream() const { return had_upstream_; }
 
 private:
+  friend class UpstreamFilterManager;
+  friend class CodecFilter;
```

Comment:
Emm...this is related to my question at https://github.com/envoyproxy/envoy/pull/22553/files/e755fee044c1ca990736a3ed605dfb729d6052d4#r945441918

Seems the filter shouldn't reference to the UpstreamRequest, it should use the filter manager callback.

Reply:
Agreed!  the filter ideally wouldn't reference the upstream request, but fixing that requires giving upstream filters access to an interface which allows them upstream-specific information.  I've got a TODO to clean up the codec filter so it can be created by the factory, and to accomplish that I have to remove all direct access, and can remove this friend class.    It's a lot more complicated code so I planned on adding it as a follow up.